cmds/core/wget/wget.go

```diff
@@ -32,42 +32,41 @@ import (
 	"net/url"
 	"os"
 	"path"
+	"strings"
 
 	"github.com/u-root/u-root/pkg/curl"
 	"github.com/u-root/u-root/pkg/uio"
 )
 
 var outPath = flag.String("O", "", "output file")
+var errEmptyURL = errors.New("empty url")
 
-func usage() {
-	log.Printf("Usage: %s [ARGS] URL\n", os.Args[0])
-	flag.PrintDefaults()
-	os.Exit(2)
+type command struct {
+	url        string
+	outputPath string
 }
 
-func run() (reterr error) {
-	log.SetPrefix("wget: ")
-
-	if flag.Parse(); flag.NArg() != 1 {
-		usage()
+func newCommand(outPath string, url string) *command {
```

Comment:
note that you can say 
output, url string
gofmt -s may or may not simplify this or complain about it ...